cypress/e2e/functional/researcher_anonymous_classes.spec.js

```diff
@@ -0,0 +1,149 @@
+    researchClasses.clickSortIcon(1);
+    researchClasses.verifyFirstRowData(1, "Cypress AutomationTeacher3");
+    researchClasses.clickSortIcon(1);
+    researchClasses.verifyFirstRowData(1, "Teacher School");
+    researchClasses.clickSortIcon(3);
+    researchClasses.verifyFirstRowData(3, "Concord Consortium");
+    researchClasses.clickSortIcon(3);
+    researchClasses.verifyFirstRowData(3, "Cypress Automation Test School");
+  });
+  it("Verify Reset All", () => {
+    researchClasses.getResetAllButton().should("exist");
+    researchClasses.getResetAllButton().should("contain", "Reset All");
+    researchClasses.verifySummary();
+    researchClasses.getResetAllButton().click({ force: true });
+    researchClasses.verifySummaryNotDisplayed();
+    researchClasses.getResultsTable().should("not.exist");
+  });
+});
+
+context("Researcher classes landing page", () => {
+
+  before(function() {
+    cy.visit(c.LEARN_PORTAL_BASE_URL); // Visit LEARN Portal home page
+    cy.login(c.TEACHER3_USERNAME, c.TEACHER3_PASSWORD); // Login as teacher user
+    cy.visit(c.LEARN_PORTAL_BASE_URL + '/portal/classes/228/materials?researcher=true');
+  });
+
+  after(function() {
+    cy.logout();
+    cy.clearAllCookies();
+  });
+
+  it("Verify research project in sidebar", () => {
+    researchClasses.getAssignmentPage().should("exist");
+    researchClasses.verifyHeader();
+    researchClasses.getOfferingsTable().should("exist");
+    researchClasses.verifySortIconNotDisplayed();
+    researchClasses.verifyActiveCheckboxDisabled();
+    researchClasses.verifyLockedCheckboxDisabled();
+    researchClasses.clickShowDetails();
+    researchClasses.verifyOfferingButtons();
+    researchClasses.verifyAnonymousStudentName();
+  });
+});
+
+context("Project Expiration Date", () => {
+
+  before(function() {
+    cy.visit(c.LEARN_PORTAL_BASE_URL); // Visit LEARN Portal home page
+    cy.login(c.TEACHER3_USERNAME, c.TEACHER3_PASSWORD); // Login as teacher user
+    cy.visit(c.LEARN_PORTAL_BASE_URL + '/users/78/edit');
+  });
+
+  after(function() {
+    cy.logout();
+    cy.clearAllCookies();
+  });
+
+  it("Verify project expiration date", () => {
+    researchClasses.getResearcherProjectChecbox().click({ force: true });
+    researchClasses.verifyDateInputNotShown();
+    researchClasses.getResearcherProjectChecbox().click({ force: true });
+    researchClasses.verifyDateInputShown();
+    researchClasses.getResearcherProjectDateInput().click({ force: true }).type("2024-04-01");
+    cy.wait(500);
+    researchClasses.getSaveButton().click({ force: true });;
+    researchClassesHelper.expandResearchProjects(); 
+    researchClassesHelper.getResearchProjectsOpen().should("exist");
+    researchClassesHelper.verifyProjectsNotDisplayed();
+    cy.visit(c.LEARN_PORTAL_BASE_URL + '/users/78/edit');
+    researchClasses.getResearcherProjectDateInput().click({ force: true }).clear();
+    cy.wait(500);
```

Comment:
Sigh. You know how I hate "wait" calls (here and 7 lines above); but if they're really necessary, leave them in.

Reply:
To debug I added the wait. Removed those from the code.